src/pkg/backup/details/details.go

```diff
@@ -124,17 +132,43 @@ func (b *Builder) AddFoldersForItem(folders []FolderEntry, itemInfo ItemInfo) {
 	defer b.mu.Unlock()
 
 	if b.knownFolders == nil {
-		b.knownFolders = map[string]struct{}{}
+		b.knownFolders = map[string]FolderEntry{}
 	}
 
 	for _, folder := range folders {
-		if _, ok := b.knownFolders[folder.ShortRef]; ok {
-			// Entry already exists, nothing to do.
-			continue
+		if existing, ok := b.knownFolders[folder.ShortRef]; ok {
+			// We've seen this folder before for a different item.
+			// Update the "cached" folder entry
+			folder = existing
 		}
 
-		b.knownFolders[folder.ShortRef] = struct{}{}
-		b.d.addFolder(folder)
+		// Update the folder's size and modified time
+		var (
+			itemSize     int64
+			itemModified time.Time
+		)
+
+		switch {
+		case itemInfo.Exchange != nil:
+			itemSize = itemInfo.Exchange.Size
+			itemModified = itemInfo.Exchange.Modified
+
+		case itemInfo.OneDrive != nil:
+			itemSize = itemInfo.OneDrive.Size
+			itemModified = itemInfo.OneDrive.Modified
+
+		case itemInfo.SharePoint != nil:
+			itemSize = itemInfo.SharePoint.Size
+			itemModified = itemInfo.SharePoint.Modified
+		}
```

Comment:
since these are all common fields for ItemInfo, maybe we should implement `Size()` and `Modified()` functions there so that this unrelated code doesn't need to be updated for new services/data categories

Reply:
That makes sense. I'm going to merge these 2 PRs and follow up with that right after today.